frontend/src/stimulus/controllers/beforeunload.controller.ts

```diff
@@ -0,0 +1,75 @@
+import { ApplicationController } from 'stimulus-use';
+import { TurboBeforeVisitEvent } from '@hotwired/turbo';
+
+export function setPageWasEdited() {
+  const application = window.Stimulus;
+  const controller = application.getControllerForElementAndIdentifier(document.body, 'beforeunload') as BeforeunloadController;
+  if (controller) {
+    controller.pageWasEdited = true;
+  }
+}
+
+export function setPageWasSubmitted() {
+  const application = window.Stimulus;
+  const controller = application.getControllerForElementAndIdentifier(document.body, 'beforeunload') as BeforeunloadController;
+  if (controller) {
+    controller.pageWasEdited = true;
+  }
+}
+
+export class BeforeunloadController extends ApplicationController {
+  /** Globally setable variable whether the page was edited */
+  public pageWasEdited = false;
+
+  /** Globally setable variable whether the page form is submitted.
+   * Necessary to avoid a data loss warning on beforeunload */
+  public pageIsSubmitted = false;
+
+  private boundBeforeUnloadHandler = this.beforeunloadHandler.bind(this);
+
+  private submitHandler = () => {
+    this.pageIsSubmitted = true;
+  };
+
+  private turboHandler = () => {
+    this.pageWasEdited = false;
+    this.pageIsSubmitted = false;
+  };
+
+  connect() {
+    super.connect();
+
+    window.addEventListener('beforeunload', this.boundBeforeUnloadHandler);
+    document.addEventListener('turbo:before-visit', this.boundBeforeUnloadHandler);
```

Comment:
Since we're handling multiple events with a single handler, why not make `BeforeunloadController` implement `EventListenerObject`? This also means there is no need for the boilerplate binding `this`.

```suggestion
  connect() {
    super.connect();
    
    this.abortController = new AbortController();
    const { signal } = this.abortController;

    window.addEventListener('beforeunload', this, { signal });
    document.addEventListener('turbo:before-visit', this, { signal });
```

Reply:
Nice 👍